python/cudf/cudf/core/scalar.py

```diff
@@ -18,6 +22,24 @@
 )
 
 
+@functools.lru_cache(maxsize=128)
+def pa_scalar_to_plc_scalar(pa_scalar: pa.Scalar) -> plc.Scalar:
```

Comment:
Is this intended to be public?

Reply:
I had thought that `cudf.core` was essentially "private" as `cudf.api` or objects in the top level `cudf` are usually public. I can lead with an `_` later if we find other public consumers

Reply:
The current module names were originally chosen to reflect pandas module. pandas does not consistently use underscores to reflect internal modules, IIRC instead pandas chooses to define `__all__` as the arbiter of this. If people rely on importing those "internal" pandas modules then we will probably feel some pain in trying to rename them, but we could do so if we thought it was worthwhile. The remapping in cudf.pandas will need to be comprehensive in that case, though.